step-automation-packages/step-automation-packages-yaml/src/main/java/step/automation/packages/yaml/AutomationPackageDescriptorReader.java

```diff
@@ -0,0 +1,143 @@
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import step.artefacts.handlers.JsonSchemaValidator;
+import step.automation.packages.AutomationPackageReadingException;
+import step.automation.packages.yaml.deserialization.YamlKeywordDeserializer;
+import step.automation.packages.yaml.model.AutomationPackageDescriptorYaml;
+import step.automation.packages.model.AutomationPackageKeyword;
+import step.automation.packages.yaml.model.AutomationPackageFragmentYaml;
+import step.core.accessors.DefaultJacksonMapperProvider;
+import step.core.dynamicbeans.DynamicValue;
+import step.core.yaml.deserializers.YamlDynamicValueDeserializer;
+import step.plans.parser.yaml.YamlPlanReader;
+import step.plans.parser.yaml.model.YamlPlanVersions;
+import step.plans.parser.yaml.schema.YamlPlanValidationException;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+
+public class AutomationPackageDescriptorReader {
+
+    private static final Logger log = LoggerFactory.getLogger(AutomationPackageDescriptorReader.class);
+
+    private final ObjectMapper yamlObjectMapper;
+
+    private final YamlPlanReader planReader;
+
+    private String jsonSchema;
+
+    public AutomationPackageDescriptorReader(String jsonSchema) {
+        // TODO: we need to find a way to resolve the actual json schema (controller config) depending on running server instance (EE or OS)
+        // TODO: also we have to resolve the json version for plans according to the automation package version!
+        this.planReader = new YamlPlanReader(null, YamlPlanVersions.ACTUAL_VERSION, false, null);
+        this.yamlObjectMapper = createYamlObjectMapper();
+
+        if (jsonSchema != null) {
+            this.jsonSchema = readJsonSchema(jsonSchema);
+        }
+    }
+
+    public AutomationPackageDescriptorYaml readAutomationPackageDescriptor(InputStream yamlDescriptor) throws AutomationPackageReadingException {
+        log.info("Reading automation package descriptor...");
+        return readAutomationPackageYamlFile(yamlDescriptor, AutomationPackageDescriptorYaml.class);
+    }
+
+    public AutomationPackageFragmentYaml readAutomationPackageFragment(InputStream yamlFragment, String fragmentName) throws AutomationPackageReadingException {
+        log.info("Reading automation package descriptor fragment ({})...", fragmentName);
+        return readAutomationPackageYamlFile(yamlFragment, AutomationPackageFragmentYaml.class);
+    }
+
+    protected <T extends AutomationPackageFragmentYaml> T readAutomationPackageYamlFile(InputStream yaml, Class<T> targetClass) throws AutomationPackageReadingException {
+        try {
+            String yamlDescriptorString = new String(yaml.readAllBytes(), StandardCharsets.UTF_8);
+
+            if (jsonSchema != null) {
```

Comment:
@jeromecomte 
This validation can be avoided by passing the jsonSchema=null to the constructor, but I still don't see the ideal way to configure the schema to be used: OS, EE or null.
For now I've added the `AutomationPackageJsonSchemaPlugin` to configure the actual json schema for local run (and in EE this plugin is overriden by another one redefiing the schema to EE).
But as you said, it doesn't cover all required cases (like desactivation).

Reply:
@iegorov777 Not sure to get the point. Adding this to the agenda of today's meeting

Reply:
@iegorov777 As discussed, a new field "schema" should be added to the automation package descriptor. It would specify the schema of the descriptor (os or ee). This will be implemented after merging this PR